官方 curl -fsSL https://claude.ai/install.sh | bash 脚本， 插件 ai 编辑运行错误哦

### What happened?

[2026-01-05T06:48:21.325Z] [INFO] Claude Code Workflow Studio is now active
[2026-01-05T06:48:21.326Z] [INFO] Claude Code Workflow Studio: All commands and handlers registered
[2026-01-05T06:48:21.408Z] [INFO] BM25 Cleanup: No legacy index data found
[2026-01-05T06:49:54.689Z] [INFO] Workflow Preview custom editor opened
  Data: {
  "fileName": "gen-img.json"
}
[2026-01-05T06:50:40.807Z] [INFO] Workflow Preview custom editor opened
  Data: {
  "fileName": "gen-img.json"
}
[2026-01-05T06:59:10.828Z] [INFO] Workflow refinement request received
  Data: {
  "requestId": "refine-1767596350824-0.8756198411977122",
  "workflowId": "workflow-1767596345658",
  "messageLength": 2,
  "currentIteration": 0,
  "maxIterations": 20,
  "useSkills": true,
  "timeoutMs": 0,
  "targetType": "workflow",
  "model": "sonnet",
  "allowedTools": [
    "Read",
    "Grep",
    "Glob",
    "WebSearch",
    "WebFetch",
    "TodoWrite"
  ]
}
[2026-01-05T06:59:10.829Z] [INFO] Starting workflow refinement
  Data: {
  "requestId": "refine-1767596350824-0.8756198411977122",
  "workflowId": "workflow-1767596345658",
  "messageLength": 2,
  "historyLength": 0,
  "currentIteration": 0,
  "useSkills": true,
  "timeoutMs": 0,
  "schemaFormat": "toon",
  "promptFormat": "toon",
  "collectMetrics": false
}
[2026-01-05T06:59:10.843Z] [INFO] Skills scanned successfully
  Data: {
  "requestId": "refine-1767596350824-0.8756198411977122",
  "personalCount": 0,
  "projectCount": 10,
  "totalCount": 10
}
[2026-01-05T06:59:10.844Z] [INFO] Skills filtered by relevance
  Data: {
  "requestId": "refine-1767596350824-0.8756198411977122",
  "filteredCount": 0,
  "topSkills": []
}
[2026-01-05T06:59:10.844Z] [INFO] Workflow schema loaded successfully
  Data: {
  "requestId": "refine-1767596350824-0.8756198411977122",
  "format": "toon",
  "sizeBytes": 21294
}
[2026-01-05T06:59:10.845Z] [INFO] Constructing refinement prompt
  Data: {
  "promptFormat": "toon",
  "schemaFormat": "toon",
  "userMessageLength": 2,
  "conversationHistoryLength": 0,
  "filteredSkillsCount": 0
}
[2026-01-05T06:59:10.847Z] [INFO] Refinement prompt constructed
  Data: {
  "promptFormat": "toon",
  "promptSizeChars": 24916,
  "estimatedTokens": 6229
}
[2026-01-05T06:59:10.847Z] [INFO] Starting Claude Code CLI streaming execution
  Data: {
  "promptLength": 24916,
  "timeoutMs": 0,
  "model": "sonnet",
  "modelName": "sonnet",
  "allowedTools": [
    "Read",
    "Grep",
    "Glob",
    "WebSearch",
    "WebFetch",
    "TodoWrite"
  ],
  "cwd": "/Users/bytedance/Demo/ask"
}
[2026-01-05T06:59:10.850Z] [INFO] Claude CLI check: using npx fallback
[2026-01-05T06:59:10.850Z] [INFO] Registered active streaming process for requestId: refine-1767596350824-0.8756198411977122
[2026-01-05T06:59:10.851Z] [INFO] Removed active streaming process (error) for requestId: refine-1767596350824-0.8756198411977122
[2026-01-05T06:59:10.851Z] [ERROR] Claude Code CLI streaming error caught
  Data: {
  "errorType": "object",
  "errorConstructor": "SubprocessError",
  "executionTimeMs": 4,
  "accumulatedLength": 0,
  "stderr": "",
  "stdout": "",
  "errorMessage": "Command failed: npx claude -p - --output-format stream-json --verbose --model sonnet --tools 'Read,Grep,Glob,WebSearch,WebFetch,TodoWrite' --allowed-tools 'Read,Grep,Glob,WebSearch,WebFetch,TodoWrite'"
}
[2026-01-05T06:59:10.851Z] [ERROR] Refinement failed during CLI execution
  Data: {
  "requestId": "refine-1767596350824-0.8756198411977122",
  "errorCode": "UNKNOWN_ERROR",
  "errorMessage": "Generation failed - please try again or rephrase your description",
  "executionTimeMs": 4
}
[2026-01-05T06:59:10.852Z] [ERROR] Workflow refinement failed
  Data: {
  "requestId": "refine-1767596350824-0.8756198411977122",
  "workflowId": "workflow-1767596345658",
  "errorCode": "UNKNOWN_ERROR",
  "errorMessage": "Generation failed - please try again or rephrase your description",
  "executionTimeMs": 4
}

### Steps to reproduce

1. Open VScode
2. 

<img width="325" height="651" alt="Image" src="https://github.com/user-attachments/assets/d9133416-812d-4b14-9315-64d9dba65158" />

### Expected behavior

希望帮忙解决

### Extension version

3.12.1

### VSCode version

_No response_

### Operating System

macOS

### Relevant log output

```shell

```

### Screenshots

<img width="1631" height="1007" alt="Image" src="https://github.com/user-attachments/assets/36841d92-3170-41f1-80bd-0891464aa239" />



which -a claude                                                                                                                                                                                             

/Users/xxx/.local/bin/claude

--- a/src/extension/services/claude-code-service.ts
+++ b/src/extension/services/claude-code-service.ts
@@ -9,6 +9,9 @@
  */
 
 import type { ChildProcess } from 'node:child_process';
+import * as fs from 'node:fs';
+import * as os from 'node:os';
+import * as path from 'node:path';
 import nanoSpawn from 'nano-spawn';
 import type { ClaudeModel } from '../../shared/types/messages';
 import { log } from '../extension';
@@ -59,36 +62,112 @@ const spawn =
 const activeProcesses = new Map<string, { subprocess: Subprocess; startTime: number }>();
 
 /**
- * Check if claude command is directly available in PATH
- * Checks every time to handle dynamic PATH changes
+ * Known Claude CLI installation paths
+ * These are checked explicitly to handle cases where VSCode Extension Host
+ * doesn't have the user's shell PATH settings (e.g., when launched from GUI)
  *
- * @returns true if claude is available, false if npx should be used
+ * Issue #375: https://github.com/breaking-brake/cc-wf-studio/issues/375
  */
-async function isClaudeCommandAvailable(): Promise<boolean> {
+const CLAUDE_KNOWN_PATHS = [
+  // Native install (macOS/Linux/WSL) - curl -fsSL https://claude.ai/install.sh | bash
+  path.join(os.homedir(), '.local', 'bin', 'claude'),
+  // Homebrew (Apple Silicon Mac)
+  '/opt/homebrew/bin/claude',
+  // Homebrew (Intel Mac) / npm global default
+  '/usr/local/bin/claude',
+  // npm custom prefix (common configuration)
+  path.join(os.homedir(), '.npm-global', 'bin', 'claude'),
+];
+
+/**
+ * Find Claude CLI executable in known installation paths
+ *
+ * @returns Full path to claude executable if found, null otherwise
+ */
+function findClaudeCliInKnownPaths(): string | null {
+  for (const p of CLAUDE_KNOWN_PATHS) {
+    if (fs.existsSync(p)) {
+      log('DEBUG', 'Found Claude CLI at known path', { path: p });
+      return p;
+    }
+  }
+  return null;
+}
+
+/**
+ * Cached Claude CLI path
+ * undefined = not checked yet
+ * null = not found (use npx fallback)
+ * string = path to claude executable
+ */
+let cachedClaudePath: string | null | undefined;
+
+/**
+ * Get the path to Claude CLI executable
+ * First checks known installation paths, then falls back to PATH lookup
+ *
+ * @returns Path to claude executable ('claude' for PATH, full path for known locations, null for npx fallback)
+ */
+async function getClaudeCliPath(): Promise<string | null> {
+  // Return cached result if available
+  if (cachedClaudePath !== undefined) {
+    return cachedClaudePath;
+  }
+
+  // 1. Check known installation paths first (handles GUI-launched VSCode)
+  const knownPath = findClaudeCliInKnownPaths();
+  if (knownPath) {
+    try {
+      const result = await spawn(knownPath, ['--version'], { timeout: 5000 });
+      log('INFO', 'Claude CLI found at known path', {
+        path: knownPath,
+        version: result.stdout.trim().substring(0, 50),
+      });
+      cachedClaudePath = knownPath;
+      return knownPath;
+    } catch (error) {
+      // Path exists but execution failed - log and continue to PATH check
+      log('WARN', 'Claude CLI found but not executable at known path', {
+        path: knownPath,
+        error: error instanceof Error ? error.message : String(error),
+      });
+    }
+  }
+
+  // 2. Fall back to PATH lookup (terminal-launched VSCode or other installations)
   try {
     const result = await spawn('claude', ['--version'], { timeout: 5000 });
-    log('INFO', 'Claude CLI check: using claude directly', {
+    log('INFO', 'Claude CLI found in PATH', {
       version: result.stdout.trim().substring(0, 50),
     });
-    return true;
+    cachedClaudePath = 'claude';
+    return 'claude';
   } catch {
-    log('INFO', 'Claude CLI check: using npx fallback');
-    return false;
+    log('INFO', 'Claude CLI not found, will use npx fallback');
+    cachedClaudePath = null;
+    return null;
   }
 }
 
 /**
+ * Clear Claude CLI path cache (for testing purposes)
+ */
+export function clearClaudeCliPathCache(): void {
+  cachedClaudePath = undefined;
+}
+
+/**
  * Get the command and args for spawning Claude CLI
- * Uses 'claude' directly if available, otherwise falls back to 'npx claude'
+ * Uses claude directly if available (from known paths or PATH), otherwise falls back to 'npx claude'
  *
  * @param args - CLI arguments (without 'claude' command itself)
  * @returns command and args for spawn
  */
 async function getClaudeSpawnCommand(args: string[]): Promise<{ command: string; args: string[] }> {
-  const useDirectClaude = await isClaudeCommandAvailable();
+  const claudePath = await getClaudeCliPath();
 
-  if (useDirectClaude) {
-    return { command: 'claude', args };
+  if (claudePath) {
+    return { command: claudePath, args };
   }
   return { command: 'npx', args: ['claude', ...args] };
 }

